modules/renter/contractor/dependencies.go

```diff
@@ -68,32 +67,43 @@ type walletBridge struct {
 func (ws *walletBridge) NextAddress() (types.UnlockConditions, error) { return ws.w.NextAddress() }
 func (ws *walletBridge) StartTransaction() transactionBuilder         { return ws.w.StartTransaction() }
 
-// stdPersist implements the persister interface via persist.SaveFile and
-// persist.LoadFile. The metadata and filename required by these functions is
-// internal to stdPersist.
+// stdPersist implements the persister interface via jj.OpenJournal and
+// jj.CheckPoint. The filename required by these functions is internal to
+// stdPersist.
 type stdPersist struct {
-	meta     persist.Metadata
+	journal  *journal
 	filename string
 }
 
-func (p *stdPersist) save(data contractorPersist) error {
-	return persist.SaveFile(p.meta, data, p.filename)
+func (p stdPersist) save(data contractorPersist) error {
+	if p.journal == nil {
+		var err error
+		p.journal, err = openJournal(p.filename, data)
+		return err
+	}
+	return p.journal.checkpoint(data)
 }
 
-func (p *stdPersist) saveSync(data contractorPersist) error {
-	return persist.SaveFileSync(p.meta, data, p.filename)
+func (p *stdPersist) update(us ...interface{}) error {
+	var updates []journalUpdate
+	for i := 0; i < len(us); i += 2 {
+		updates = append(updates, newJournalUpdate(us[i].(string), us[i+1].(interface{})))
+	}
+	return p.journal.update(updates)
 }
 
 func (p *stdPersist) load(data *contractorPersist) error {
-	return persist.LoadFile(p.meta, data, p.filename)
+	var err error
+	p.journal, err = openJournal(p.filename, data)
+	if err != nil {
+		// try loading old persist
+		return loadv110persist(filepath.Dir(p.filename), data)
```

Comment:
what I would really like to do is move all the existing COMPAT code out of `load` and into `loadv110persist`. But there's a dilemma: the COMPAT code requires access to the contractor (since it does stuff like rescanning), so it can't happen in this method. But it also can't happen in `c.load`, because `c.load` doesn't have access to the persist directory. (It is hidden by the `persister` dependency.)